src/frontend/lib/models/perpetrator_model.dart

```diff
@@ -0,0 +1,75 @@
+// A class that represents either a Pest or a Disease.
+import 'package:cloud_firestore/cloud_firestore.dart';
+
+class PerpetratorModel {
```

Comment:
I like the fact that you dedicated this model to pests and diseases, as we wanna store different metadata. However, what model will we use for non-perpetrator insights such as nutrient deficiencies now? Wouldn't that also come with a reference image and description at least?

Reply:
If they share the same structure we can make it even more general. I think me and Paul forgot about the fact those need images too

Reply:
Nutrient deficiencies would come with their own images indeed, but for something like standing water that wouldn't be the case. I would personalize the details screen for the insights per insight type, and try to introduce some sort of a view inheritance model since a lot of the data would be reusable.

Reply:
I see, I'd suggest make the perpetrator model extend the more general insight model? And have attributes like images and description in there?